### PR TITLE
Update resolve settings server version

### PR DESCRIPTION
**Pull request by dubdabasoduba**

### CODE REVIEW

File: opensrp-app/src/main/java/org/smartregister/SyncConfiguration.java [hunk 1]
```diff
@@ -152,11 +152,10 @@ public boolean hasExtraSettingsSync() {
 
     /**
      * Allows the user to define extra settings sync parameters
-     *
-     * @return settingsParams {@link List<String>}
+     * @return
      */
-    public List<String> getExtraSettingsParameters() {
-        return new ArrayList<>();
+    public String getExtraSettingsParameters() {
```
Comment (ndegwamartin) [inline opensrp-app/src/main/java/org/smartregister/SyncConfiguration.java:157]: Do we maybe want to overload this public method for backward compatibility and mark the former one with an annotation for `@Deprecated`?

File: opensrp-app/src/main/java/org/smartregister/sync/helper/SyncSettingsServiceHelper.java [hunk 2]
```diff
@@ -89,24 +89,22 @@ protected CoreLibrary getInstance() {
...
-                    String url = SettingsSyncIntentService.SETTINGS_URL + "?" + params + "&" + AllConstants.SERVER_VERSION + "=" + sharedPreferences.fetchLastSettingsSyncTimeStamp() + "&" + AllConstants.RESOLVE + "=" + getInstance().getSyncConfiguration().resolveSettings();
-                    JSONArray extraSettings = pullSettings(url, accessToken);
-                    if (extraSettings != null) {
-                        aggregateSettings(completeExtraSettings, extraSettings);
-                    }
-                }
-
-                aggregateSettings(settings, completeExtraSettings);
+            String syncParams = getInstance().getSyncConfiguration().getExtraSettingsParameters();
+            String url = SettingsSyncIntentService.SETTINGS_URL + "?" + syncParams + "&" + AllConstants.SERVER_VERSION + "=0&" + AllConstants.RESOLVE + "=" + getInstance().getSyncConfiguration().resolveSettings();
```
Comment (ndegwamartin) [inline opensrp-app/src/main/java/org/smartregister/sync/helper/SyncSettingsServiceHelper.java:97]: Also check out if the [Request Params Builder class](https://github.com/opensrp/opensrp-client-core/blob/ec663ec9c86beeea92819517f271f725a4bcc44f/opensrp-app/src/main/java/org/smartregister/sync/intent/BaseSyncIntentService.java#L37) in the `BaseSyncIntentService` would be helpful here

File: opensrp-app/src/main/java/org/smartregister/SyncConfiguration.java (hunk 1 above)
Comment (dubdabasoduba) [reply opensrp-app/src/main/java/org/smartregister/SyncConfiguration.java:157]: I will create another project. we can not overload coz of the different retutn types

File: opensrp-app/src/main/java/org/smartregister/SyncConfiguration.java [hunk 3]
```diff
@@ -152,8 +152,20 @@ public boolean hasExtraSettingsSync() {
 
     /**
      * Allows the user to define extra settings sync parameters
+     * This returns the parameters as a string
      *
-     * @return settingsParams {@link List<String>}
+     * @return
+     */
+    public String getExtraStringSettingsParameters() {
```
[the remark this answers is not in the document]
Comment (ndegwamartin) [reply opensrp-app/src/main/java/org/smartregister/SyncConfiguration.java:159]: We might need to document on the README.md or Wiki how to use/configure these new settings configuration(s)

**Review by ndegwamartin — APPROVED** [review]

**Pull request merged**